arangod/RocksDBEngine/RocksDBEngine.cpp

```diff
@@ -4162,4 +4207,63 @@ void RocksDBEngine::addCacheMetrics(uint64_t initial, uint64_t effective,
     _metricsEdgeCacheEmptyInserts.count(totalEmptyInserts);
   }
 }
-}  // namespace arangodb
\ No newline at end of file
+
+using SortingMethod = arangodb::basics::VelocyPackHelper::SortingMethod;
+
+Result RocksDBEngine::writeSortingFile(SortingMethod sortingMethod) {
+  auto& databasePathFeature = server().getFeature<DatabasePathFeature>();
+  std::string path = databasePathFeature.subdirectoryName(kSortingMethodFile);
+  std::string value =
+      sortingMethod == SortingMethod::Legacy ? "LEGACY" : "CORRECT";
+  try {
+    basics::FileUtils::spit(path, value, true);
+  } catch (std::exception const& ex) {
+    LOG_TOPIC("8ff0f", ERR, Logger::STARTUP)
+        << "unable to write 'SORTING' file '" << _path << "': " << ex.what()
+        << ". Please make sure the file/directory is writable for the "
+           "arangod process and user!";
+    return {TRI_ERROR_CANNOT_WRITE_FILE};
+  }
+  return {};
+}
+
+SortingMethod RocksDBEngine::readSortingFile() {
+  SortingMethod sortingMethod = SortingMethod::Legacy;
+  auto& databasePathFeature = server().getFeature<DatabasePathFeature>();
+  std::string path = databasePathFeature.subdirectoryName(kSortingMethodFile);
+  std::string value;
+  try {
+    basics::FileUtils::slurp(path, value);
+    value = arangodb::basics::StringUtils::trim(value);
+    sortingMethod =
+        (value == "LEGACY") ? SortingMethod::Legacy : SortingMethod::Correct;
+  } catch (std::exception const& ex) {
+    // When we see a database directory without SORTING file, we fall back
+    // to legacy mode, except for agents. Since agents have never used
```

Comment:
see the comment above for agents' use of vpack indexes.

Reply:
See above.